### PR TITLE
chore: update x/net to v0.32

### PR DESCRIPTION
**Pull request by Emyrk**
Tailscale fix: https://github.com/tailscale/tailscale/pull/14297
Golang Issue: https://github.com/golang/go/issues/70528
Commit in x/net that fixes it: https://github.com/golang/net/commit/e9cd716925e5c6cfb200c82f9d9bf6bc4118d47a
Coder issue: https://github.com/coder/coder/issues/15616

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -75,16 +75,16 @@ require (
 	go.uber.org/zap v1.24.0
 	go4.org/mem v0.0.0-20220726221520-4f986261bf13
 	go4.org/netipx v0.0.0-20230728180743-ad4cb58a6516
-	golang.org/x/crypto v0.18.0
+	golang.org/x/crypto v0.30.0
```
Comment (spikecurtis) [inline go.mod:78]: I think there is a CVE in 0.30.0 --- can we bump to 0.31.0?

https://go.googlesource.com/crypto/+/refs/tags/v0.31.0

File: tailcfg/tailcfg_test.go [hunk 2]
```diff
@@ -571,6 +571,17 @@ func TestNetInfoFields(t *testing.T) {
...
+		"DERPLatencyV6",
+		"UDP",
+		"IPv6",
+		"IPv4",
+		"IPv6CanSend",
+		"IPv4CanSend",
+		"ICMPv4",
+		"GlobalV4",
+		"GlobalV6",
+		"CaptivePortal",
```
Comment (spikecurtis) [inline tailcfg/tailcfg_test.go:584]: why is this change needed?

File: go.mod (hunk 1 above)
Comment (Emyrk) [reply go.mod]: Done :+1:

File: tailcfg/tailcfg_test.go (hunk 2 above)
Comment (Emyrk) [reply tailcfg/tailcfg_test.go:584]: If you run `go generate ./...`, you get this test fails: https://github.com/coder/tailscale/blob/d8574de4b4ca468d8c631be581d095650b31426f/tailcfg/tailcfg_test.go#L559-L559

The change is only in a test, that validates the `Clone()` method is missing some fields. These fields were omitted in the test that we added here: https://github.com/coder/tailscale/pull/56

I was just getting a test to pass that was failing locally.

**Review by spikecurtis — APPROVED** [review]
